papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -0,0 +1,563 @@
+Similarly, aggregating over replicas (e.g. the same simulation with different random seeds) facilitates computing averaged quantities and error bars.
+Another example is submitting aggregates with a fixed number of jobs in each aggregate to enable massive parallelization by breaking a large MPI communicator into a smaller communicator for each independent job, which is necessary for efficient utilization of leadership-class supercomputers like OLCF Summit.
+
+Finally, bundling is another way to use workflows in conjunction with an HPC scheduling system.
+Whereas aggregates are concerned with jobs and groups operations, bundling is concerned with combining executable units into a single submission script.
+This distinction means that bundling is not part of the workflow definition, but a means of tailoring batch scripts for different HPC systems.
+Bundles allow users to leverage scheduler resources effectively and minimize queue time, and can be run in serial (the default) or parallel.
+Users enable bundling by passing the command line argument ``--bundle``, optionally with another argument ``--parallel`` to run each command in the bundle in parallel (the Python API has corresponding options as well).
+The simplest case of a bundle is a submission script with the same operation being executed for multiple jobs.
+Bundling is what allows the submission script to contain multiple jobs executing the same operation.
+By storing information about the generated bundles during submission, **signac-flow** prevents accidental resubmission just as in the unbundled case.
+While the example above does not use either groups or aggregation, bundles works seamlessly with both.
+
+Synced Collections: Backend-agnostic, persistent, mutable data structures
+-------------------------------------------------------------------------
+
+Motivation
+~~~~~~~~~~
+
+At its core, **signac** is a tool for organizing and working with data on the filesystem, presenting a Pythonic interface for tasks like creating directories and modifying files.
+In particular, **signac** makes modifying the JSON files used to store the state points and documents as easy as working with Python dictionaries.
+Despite heavily optimizing these APIs, in seeking to scale **signac** to ever-larger data spaces we quickly realized that the most significant barrier intrinsic to our model was the overhead of parsing and modifying large numbers of text files.
+Unfortunately, the usage of JSON files in this manner was deeply embedded in our data model, making switching to a more performant backend without breaking APIs or severely complicating our data model a daunting task.
+
+While attempting to overcome this hurdle, we identified a core problem that our current API for modifying JSON files was solving in a limited manner: namely, the creation of a dictionary-like interface to an underlying resource.
```

Comment:
it's not clear what "this hurdle" is, especially since it's the start of a new paragraph.  Do you mean attempting to be less reliant on json files?

Reply:
```suggestion
While attempting to overcome the code's ties to individual JSON files, we identified a core problem that our current API for modifying JSON files was solving in a limited manner: namely, the creation of a dictionary-like interface to an underlying resource.
```

Reply:
```suggestion
While attempting to overcome the code's ties to individual JSON files, we identified a core problem in our current API. The modification of JSON files was solved in a limited manner: namely, the creation of a dictionary-like interface to an underlying resource.
```

Could we also break this sentence up?

Reply:
The new format suggests that that there was a problem in a negative sense whereas before it was merely describing a domain of application namely interfacing to an underlying resource through a data-structure like approach.

Reply:
How about something like:

While attempting to overcome the code's ties to individual JSON files, we identified a core problem that our current API for modifying JSON files. The problem was solved in a limited manner: namely, the creation of a dictionary-like interface to an underlying resource.

Sorry about the format. I can't find the suggestions button.

Reply:
I thought the hurdle was clear, but we can make it clearer. I rewrote it so that the tone is different as well.
```suggestion
While attempting to separate the **signac** data model from its original backend implementation (manipulating JSON files on disk), we identified a common pattern: providing a dictionary-like interface for an underlying resource.
```